src/helpers/UserStatus.js

```diff
@@ -1,19 +1,26 @@
 // @flow
 import * as React from 'react';
 
-import { withUser } from '../context/User';
+import { withUser, withSimpleToken } from '../context/User';
 import type { User } from '../types';
 
 type Props = {
   user: User,
+  simpleToken: ?string,
   children?: React.Node,
 };
 
-const RequireUserCriteria = (user_criteria: (user: User) => boolean) =>
-  withUser((props: Props) => {
-    if (!user_criteria(props.user)) return null;
-    return props.children;
-  });
+const RequireUserCriteria = (
+  user_criteria: (user: User | ?string) => boolean,
+) => {
+  return withSimpleToken(
+    withUser((props: Props) => {
+      if (!user_criteria(props.user) && !user_criteria(props.simpleToken))
```

Comment:
I don't get this part, is even `user_criteria` needed?

looks like `if (!props.user && !user.simpleToken)` would do the job